### PR TITLE
Build all 6 debian packages & Add debian yum publish script

### PR DESCRIPTION
**Pull request by derekbekoe**
Closes https://github.com/Azure/azure-cli/issues/3720

### CODE REVIEW

**Comment by promptws** [comment]
View a preview at https://prompt.ws/r/Azure/azure-cli/5470
This is an **experimental preview** for @microsoft users.

File: build_scripts/debian/dir_creator.sh [hunk 1]
```diff
@@ -90,9 +90,15 @@ THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLI
 
 EOM
 
+# debian/azure-cli/opt/az/lib/python3.6/lib-dynload/_ssl.cpython-36m-x86_64-linux-gnu.so
```
Comment (troydai) [inline build_scripts/debian/dir_creator.sh]: Should we just remove this line?

File: tools/automation/clibuild/__init__.py [hunk 2]
```diff
@@ -16,18 +17,41 @@
...
+        ('wheezy', 'debian:wheezy'),
+        ('jessie', 'debian:jessie'),
+        ('stretch', 'debian:stretch'),
+        ('artful', 'ubuntu:artful'),
+        ('xenial', 'ubuntu:xenial'),
+        ('trusty', 'ubuntu:trusty'),
+        ('bionic', 'ubuntu:bionic'),
+    ]
+    with ThreadPoolExecutor(max_workers=len(dists)) as executor:
+            tasks = {executor.submit(build_debian, dist_info, git_url, git_branch, cli_version, artifact_dir, arg_ns=arg_ns) for dist_info
```
Comment (troydai) [inline tools/automation/clibuild/__init__.py]: Don't use 'tab'

File: build_scripts/debian/dir_creator.sh (hunk 1 above)
Comment (derekbekoe) [reply build_scripts/debian/dir_creator.sh]: Added a TODO to explain

File: tools/automation/clibuild/__init__.py (hunk 2 above)
Comment (derekbekoe) [reply tools/automation/clibuild/__init__.py]: huh?

**Pull request merged**